src/Microsoft.VisualStudio.Threading.Tests/TplExtensionsTests.cs

```diff
@@ -151,12 +151,47 @@ public void ApplyResultToPreCompletedNonGeneric()
         [Fact]
         public void WaitWithoutInlining()
```

Comment:
On second thought, on your No. 1 point, I think it would be interesting to document whether AttachToParent tasks end up getting inlined as well. I'll add a test.